fix(assertions): output and mapping assertions do not accept logical id (#16329)
aws/aws-cdk: packages/@aws-cdk/assertions/README.md

partial object matching. This behavior can be configured using matchers.
See subsequent section on [special matchers](#special-matchers).

> ## Output and Mapping Matching

Reviewer: ```suggestion
## Output and Mapping sections
```

> The module allows you to assert that an output or matching has specific properties.

Reviewer: ```suggestion
The module allows you to assert that the CloudFormation template contains an Output that matches specific properties.
```

> The following code asserts that a resource contains an output with a `logicalId` of

Reviewer: ```suggestion
The following code asserts that a template contains an Output with a `logicalId` of
```

`Foo` and the specified properties -

```ts
assert.hasOutput('Foo', { 
  Value: 'Bar',
  Export: { Name: 'ExportBaz' }, 
});
```

> Alternatively, if you want to match the value to all outputs, you can use the `'*'`
> special case as the `logicalId`.

Reviewer: ```suggestion
If you want to match against all Outputs in the template, use `*` as the `logicalId`.
```

```ts
assert.hasOutput('*', {
  Value: 'Bar',
  Export: { Name: 'ExportBaz' },
});
```

`findOutputs()` will return a list of outputs that match the `logicalId` and `props`,

> and you can use the `'*'` special case as well. `hasMapping()` and `findMappings()`
> follow a similar pattern to output matching.

Reviewer: ```suggestion
and you can use the `'*'` special case as well. 

The APIs `hasMapping()` and `findMappings()` provide similar functionalities.
```
